### DISCUSSION TITLE
Question: No highlights in editor for formatter?

### DISCUSSION THREAD

**Discussion by DPHonys** [biomejs/biome-vscode#379] [Q&A] [answered]
### VS Code version

1.94.0

### Extension version

2.3.0

### Biome version

1.9.3

### Operating system

- [X] Windows
- [X] macOS
- [ ] Linux

### Description

So found out the WSL is not the issue.. biome works and shows errors if they are linter errors. But if they are formatter errors they are not shown in the editor. Is there any way to see the formatter errors as warnings or errors? Like with prettier. Or is this feature on a road map? 

### Steps to reproduce

1. Create new project with pnpm
2. Install biome as dev dependency
3. Install vscode extension
4. Set up biome.json in root
5. Create index.ts file with known FORMATTER issue
6. Run pnpm biome format index.ts - see formatter issue
7. In editor no warning about the formatting

### Expected behavior

I expect to see the same formatter errors in the editor as when using biome cli as warnings.

### Does this issue occur when using the CLI directly?

Not sure / Not applicable

**Comment by nhedger** [comment, marked as answer]
Hey @DPHonys,

We don't report formatting differences as diagnostics and have no plans to do it at the moment.

The recommended course of action is to setup **Format on Save** in your editor to automatically format files.

**Reply by DPHonys** [reply]
Okay, that's reasonable... But it would be awesome to have an option for it. Sometimes my VSCode doesn't run the formatter on save (Why? No idea) and I need to run the formatter manually. So when I'm using Prettier and I see the warnings after saving, I know I have to run the formatter manually... without the warnings, there's no in-editor indication of formatting errors.

I know it's probably a "me problem," but I still wanted to ask about it. So, thanks for the answer. 🐼

**Reply by timkinnane** [reply]
I feel the same as I often have trouble configuring Biome in a monorepo or where I'm migrating from another tool. One of the best ways to get quick visual feedback that Biome is working is to look for error lines in files I know won't meet the new standard. 

With other tools things like trailing commas or line endings would show red squiggles, which is a good indication at a glance that not only is the linter integration with VS Code configured correctly, but also gives some cursory feedback of how drastic the changes will be when applied on save.

With Biome there's often nothing shown, so it's hard to tell if it's not working or there's just nothing in the file that conflicts with the subset of rules that would show an error (e.g. complexity rules like `useLiteralKeys`).
